### DISCUSSION TITLE
Can we Strimzi support APISIX gateway?

### DISCUSSION THREAD

**Discussion by peter-hst** [strimzi/strimzi-kafka-operator#8694] [Ideas]
Can we Strimzi support [APISIX](http://apisix.apache.org/) gateway? Thanks a lot!

**Comment by scholzj** [comment]
Sorry, but support it for what?

**Comment by peter-hst** [comment]
Sorry to trouble you. because, we wants to using  this APISIX gateway replace ingress-nginx. It doesn't matter if you don't support it, I just consult and then we are making plans!

**Reply by scholzj** [reply]
I do not think we plan to officially support it. And I'm not even sure it is possibly given APISIX is as far as I understood mainly an API Gateway. You would need to make sure it can do things such as TLS passthrough or some kind of TCP routing etc. Without that you cannot use it since Kafka is not HTTP traffic.

If it supports the required features, you can use it always manually:
* Create `type: cluster-ip` listener in Strimzi Kafka cluster
* Create manually the required Ingress resources to make sure APISIX routes the traffic to the broker services
* Override the `advertisedHostname` fields in the listener configuration in the `Kafka` CR to set it to the addresses used by the Ingresses.

**Reply by peter-hst** [reply]
OK, I see, may be we need to try and test first, Thank you!

**Reply by peter-hst** [reply]
I saw the document, it seems support [TCP-routing](https://apisix.apache.org/blog/2021/10/09/apisix-ingress-techblog/#apisix-ingress-features-explained) I'll try, Thank you!
